src/internal/react-enhancer.ts

```diff
@@ -57,15 +57,23 @@ const reactEnhancer = <T>(
   }) as BehaviorObservable<T>
 
   let promise: any
+  let hasError = false
+  let error: unknown = undefined
   const getValue = () => {
+    if (hasError) {
+      throw error
+    }
```

Comment:
I compared with the way that this is solved in React's example and within their `read` (our `getValue()` equivalent) they either return the value, throw the promise, or throw the error.